### ISSUE TITLE
Wrong week numbers for discontinuous dates with more than two years difference

### ISSUE THREAD

**Issue by alberto-agudo** [grahamjwhite/IndexNumR#79]
Hi @grahamjwhite! 

I've been using your package for different index numbers tasks. In one of them I had discontinuous dates for some stores where the sales were resumed with more than two years difference. Even though I am not using the `priceIndex` function with these stores directly as I need some further processing to allow for the gap, I am using the `weekIndex` function to create this variable. However, it is creating weird results that have been hard for me to notice. As you can see below, for a two years difference of 72 weeks, it is saying that the diff is of 55 weeks, probably because it understands that there is no additional year in the middle.

Is it worth solving?

--- 
Code:
``` r
weeks <- structure(c(17595, 17602, 17609, 17616, 17623, 17630, 17637,
17644, 17651, 17658, 17665, 17672, 17679, 17686, 17693, 17700,
17707, 17714, 17721, 17728, 17735, 17742, 17749, 17756, 17763, 
17770, 17777, 17784, 17791, 17798, 17805, 17812, 17819, 17826,
17833, 18337, 18344, 18351, 18358, 18365, 18372, 18386, 18393,
18400, 18407, 18414, 18421, 18428, 18435, 18442, 18449, 18456,
18463, 18470, 18477, 18484, 18491, 18498, 18505, 18512, 18519
), class = "Date")


week_nr <- IndexNumR::weekIndex(weeks)
df <- data.frame(weeks, week_nr)
df[35:36, ]
#>         weeks week_nr
#> 35 2018-10-29      35
#> 36 2020-03-16      55
difftime(as.Date("2020-03-16"), as.Date("2018-10-29"), units = "weeks") 
#> Time difference of 72 weeks
```

<sup>Created on 2023-09-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by grahamjwhite** [comment]
Hi @alberto-agudo,

This is an interesting one. The weekIndex function assumes you're working with a continuous series with no gaps, which is why you're getting weird behaviour. It is missing the 52 weeks in 2019. I have an idea of how it could potentially be fixed, but I'd like to understand what the use case is. Do you need the weekIndex function to return an accurate result for your downstream processing? Or can you fix the gap first and then run weekIndex on the continuous series? 

Graham

**Comment by alberto-agudo** [comment]
Hi @grahamjwhite, 

Sorry it took so long to get back to you. Actually I solved it through `tidyr::complete`. I understand that the design use case is a 
(year) continuous time series and I may have overused the package for some data wrangling before using the `priceIndex` function. It has worked perfectly with `tidyr::complete` so I guess there's no need to make the code dirtier through a very ad-hoc and rare case. However, I did spot a bug in the `weekIndex` function that I will better explain, reopening #72.

This one is fixed, thank you for your response. 

Best, 
Alberto.
